### PR TITLE
improvement-NXBT-3668-rework-branch-jobs-to-use-branch-parameter

### PR DESCRIPTION
**Pull request by NourNuxeo**

### CODE REVIEW

**Comment by ataillefer** [comment]
Maybe remove the `defaultValue('2021')` ?

**Comment by NourNuxeo** [comment]
This is done simultaneously with:
update to integration-scripts-priv/2021 https://github.com/nuxeo/integration-scripts-priv/pull/245
update to integration-scripts-priv/2023 https://github.com/nuxeo/integration-scripts-priv/pull/246
update to nuxeo-lts/2021 https://github.com/nuxeo/nuxeo-lts/pull/646
update to nuxeo-lts/2023 https://github.com/nuxeo/nuxeo-lts/pull/647
update to nuxeo-jsf-ui-lts/2021 https://github.com/nuxeo/nuxeo-jsf-ui-lts/pull/55
update to nuxeo-jsf-ui-lts/2023 https://github.com/nuxeo/nuxeo-jsf-ui-lts/pull/56

**Comment by NourNuxeo** [comment]
I've been able to transmit the NUXEO_BRANCH param accros the following (simplified) pipelines in the staging environment
```
release nuxeo
-> release nuxeo-jsf
  -> release jira
    -> release hf
      -> publish release notes 
```

I didn't request reviewers on all the PRs to avoid notification flood, but they are all linked in my previous message.

**Review by kevinleturc — APPROVED** [review]

File: charts/jenkins/jobs/nuxeo/lts/release-nuxeo.groovy.gotmpl [hunk 1]
```diff
@@ -25,13 +25,18 @@ pipelineJob('nuxeo/lts/release-nuxeo-2021') {
...
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
       // Script Path
       scriptPath('ci/Jenkinsfiles/release.groovy')
     }
   }
   parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to promote a given Nuxeo Server build version as a release.')
```
Comment (ataillefer) [inline charts/jenkins/jobs/nuxeo/lts/release-nuxeo.groovy.gotmpl:37]: Simply "Branch to release"?

File: charts/jenkins/jobs/nuxeo/lts/release-nuxeo-jsf-ui.groovy.gotmpl [hunk 2]
```diff
@@ -25,13 +25,18 @@ pipelineJob('nuxeo/lts/2023/release-nuxeo-jsf-ui-2023') {
...
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
       // Script Path
       scriptPath('ci/Jenkinsfiles/release.groovy')
     }
   }
   parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to release Nuxeo JSF UI.')
```
Comment (ataillefer) [inline charts/jenkins/jobs/nuxeo/lts/release-nuxeo-jsf-ui.groovy.gotmpl:37]: Branch to release

File: charts/jenkins/jobs/nuxeo/lts/release-jira.groovy.gotmpl [hunk 3]
```diff
@@ -25,13 +25,18 @@ pipelineJob('nuxeo/lts/2023/release-jira-2023') {
...
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
       // Script Path
       scriptPath('Jenkinsfiles/release-jira.groovy')
     }
   }
   parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to release in Jira.')
```
Comment (ataillefer) [inline charts/jenkins/jobs/nuxeo/lts/release-jira.groovy.gotmpl:37]: Branch to release (simple and consistent with the other jobs if it's the same everywhere)

File: charts/jenkins/jobs/nuxeo/lts/release-nuxeo-hf.groovy.gotmpl [hunk 4]
```diff
@@ -25,13 +25,18 @@ pipelineJob('nuxeo/lts/2023/release-nuxeo-hf-2023') {
...
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
       // Script Path
       scriptPath('Jenkinsfiles/release-hf.groovy')
     }
   }
   parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to promote a given Nuxeo hotfix package version as a release.')
```
Comment (ataillefer) [inline charts/jenkins/jobs/nuxeo/lts/release-nuxeo-hf.groovy.gotmpl:37]: Branch to release

File: charts/jenkins/jobs/nuxeo/lts/publish-nuxeo-hf-release-notes.groovy.gotmpl [hunk 5]
```diff
@@ -25,13 +25,18 @@ pipelineJob('nuxeo/lts/publish-nuxeo-hf-release-notes-2021') {
...
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
       // Script Path
       scriptPath('Jenkinsfiles/publish-hf-release-notes.groovy')
     }
   }
   parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to update Hotfix package description with the JIRA release notes.')
```
Comment (ataillefer) [inline charts/jenkins/jobs/nuxeo/lts/publish-nuxeo-hf-release-notes.groovy.gotmpl:37]: Branch to release

**Comment by ataillefer** [comment]
Also, I guess it `pipelineJob('nuxeo/lts/release-nuxeo') {` instead of `nuxeo/lts/release-nuxeo-2021`...

**Pull request merged**